src/utils/deprecated/deprecated.tsx

```diff
@@ -3,25 +3,65 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import React, { useEffect } from 'react';
+import React, { useEffect, useRef } from 'react';
 
 export const getDeprecatedMessage = (message: string): string =>
   `[DEPRECATED] ${message}`;
 
-export const deprecated = (message: string) => {
+interface IDeprecatedComponentWarning {
+  newComponentName: string;
+  version?: string;
+  getMessage?: (deprecatedComponentName: string) => string;
+}
```

Comment:
It might be more user-friendly to have the interface look something like

```ts
{
  newComponentName: string;
  version?: string;
} | {
  getMessage: () => string;
}
```

Reply:
Good idea, I've made changes